lib/addons/fluxcd/index.ts

```diff
@@ -0,0 +1,101 @@
+// lib/fluxcd_addon.ts
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace } from "../../utils";
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
+import { FluxGitRepository, GitRepositoryProps } from "./gitrepository"
+import { Cluster } from 'aws-cdk-lib/aws-ecs';
+import { KubernetesManifest } from 'aws-cdk-lib/aws-eks/lib/k8s-manifest';
+/**
+ * User provided options for the Helm Chart
+ */
+export interface FluxCDAddOnProps extends HelmAddOnUserProps {
+  /**
+   * To Create Namespace using CDK
+   */    
+  createNamespace?: boolean;
+
+  /**
+   * Optional values for `GitRepository` Source to produce an Artifact for a Git repository revision.
+   */
+  gitRepositoryProps?: GitRepositoryProps;

```

Comment:
Can you use `bootstrapRepo?: spi.ApplicationRepository;`  exactly the same way it is done in Argo?